certbot-apache/certbot_apache/configurator.py

```diff
@@ -85,27 +87,53 @@ class ApacheConfigurator(augeas_configurator.AugeasConfigurator):
 
     description = "Apache Web Server plugin - Beta"
 
+    OS_DEFAULTS = dict(
```

Comment:
I do agree, and looked into it a bit when implementing this, but for the sake of keeping this already pretty large PR from growing I decided to leave it for later.

Reply:
> I do agree, and looked into it a bit when implementing this, but for the sake of keeping this already pretty large PR from growing I decided to leave it for later.

Great. I created #5289.

> This would and should go to the same issue and development we discussed in IRC: letting users to pick override classes from CLI if fingerprinting fails. What I also want to accomplish with this would be support for multiple Apache installations in a system. Ie. giving better UX to choose root filepath, apachectl path and so on. Because of the current list() nature of the cmds it's not trivial to provide a CLI flag.

I agree we should do that work as well. As long as the new system is designed carefully with that kind of functionality in mind, they wouldn't have to be done together to keep PRs small, but it probably doesn't matter much as the changes shouldn't be that large.

I completely agree the Apache command line flags need to be reworked. Just to state this explicitly, we should try to preserve backwards compatibility and deprecate things slowly if we make any breaking changes.